packages/shared/src/components/auth/ForgotPasswordForm.tsx

```diff
@@ -1,26 +1,68 @@
 import classNames from 'classnames';
-import React, { ReactElement, useState } from 'react';
+import React, { FormEvent, ReactElement, useState } from 'react';
+import { useMutation, useQuery } from 'react-query';
+import {
+  AccountRecoveryParameters,
+  getErrorMessage,
+  getNodeByKey,
+  getNodeValue,
+  initializeRecovery,
+  sendEmailRecovery,
+} from '../../lib/auth';
+import { formToJson } from '../../lib/form';
+import { disabledRefetch } from '../../lib/func';
 import { Button } from '../buttons/Button';
 import { TextField } from '../fields/TextField';
 import MailIcon from '../icons/Mail';
 import VIcon from '../icons/V';
 import { CloseModalFunc } from '../modals/common';
 import AuthModalHeader from './AuthModalHeader';
-import { AuthModalText } from './common';
+import { AuthForm, AuthModalText } from './common';
+import TokenInput from './TokenField';
 
 interface ForgotPasswordFormProps {
-  email?: string;
+  initialEmail?: string;
   onBack?: CloseModalFunc;
   onClose?: CloseModalFunc;
 }
 
 function ForgotPasswordForm({
-  email,
+  initialEmail,
   onBack,
   onClose,
 }: ForgotPasswordFormProps): ReactElement {
+  const [hint, setHint] = useState('');
   const [emailSent, setEmailSent] = useState(false);
-  const onSendEmail = () => setEmailSent(true);
+  const { data: recovery } = useQuery('recovery', initializeRecovery, {
+    ...disabledRefetch,
+  });
```

Comment:
We'll probably have to introduce a manual refetch based on the resend?
Just noting it here for you to test with. (could be it's multi try)

Reply:
we can probably change keys, an incrementing number state of how much the user has sent already so that it would auto-refetch. What do you think?

Reply:
I'll fix the deprecated change on merge now.

Reply:
Sure also fine, just make sure it's even needed.
If we don't need to refetch it's obviously better. *(Kratos might keep the CSRF valid)

Reply:
Yes, sometimes, the flow gets invalid after a period of time (which is tied to the token). I think we will need it.